packages/neon-core/src/tx/transaction/Transaction.ts

```diff
@@ -197,13 +197,8 @@ export class Transaction {
     return this;
   }
 
-  /**
-   * Add an attribute.
-   * @param usage The usage type. Do refer to txAttrUsage enum values for all available options.
-   * @param data The data as hexstring.
-   */
-  public addAttribute(usage: number | string, data: string): this {
-    this.attributes.push(new TransactionAttribute({ usage, data }));
+  public addAttribute(...attributes: Array<TransactionAttribute>) {
```

Comment:
lets standardise if we want to accept a TransactionAttribute, TransactionAttributeLike,  Array<TransactionAttribute> or Array<TransactionAttributeLike> here. All the methods currently have a different style. I prefer adding a single TransactionAttributeLike.

Reply:
I dont really mind having the spread param but you will have to update the method name to use the plural form.

Reply:
I have changed all add methods to accept `Spread Array` and `Like`, I don't think the method name should be plural, as spread param method name should not be plural

Reply:
The methods on the builder are all spread params and in plural form. 

My argument against spread params is that it will be difficult to adjust the api in the future if there is a need to introduce or tweak an argument. The spread param also requires annotations on the API itself. It is not clear that the method provides the capability to add multiple items if the name is singular.

Reply:
OK. Base class interfaces better be simple.
I changed to single param and removed unnecessary logic for `addWitness`, please check.